lib/Doctrine/DBAL/Platforms/PostgreSqlPlatform.php

```diff
@@ -1186,4 +1188,23 @@ public function quoteStringLiteral($str)
 
         return parent::quoteStringLiteral($str);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefaultValueDeclarationSQL($field)
+    {
+        if (self::isSerialField($field) && empty($field['notnull'])) {
```

Comment:
This condition will result to `true` when `$field['notnull']` doesn't exist, is this the expected behaviour? I'd suggest to be more explicit about that, like `$field['notnull'] ?? false === false`.

Reply:
I know that we use `empty()` on the parent method, but we didn't have these features back then.

Reply:
Is there any reason to leave the `empty($field['notnull'])` out of the method?

Reply:
Also do we have unit tests covering all the paths?

Reply:
> Is there any reason to leave the empty($field['notnull']) out of the method?

Yep. `nonnull` doesn't have anything to do with being a `SERIAL` field. Also: this is perfectly okay: `whatever SERIAL NOT NULL`. So if someone wants to make sure the field is `notnull` they are free to do so.

> This condition will result to true when $field['notnull'] doesn't exist, is this the expected behaviour?

`$field['notnull']` missing is the same as `$field['notnull'] = false` yeah? I think you're right though, should be more explicit.

> Also do we have unit tests covering all the paths?

I need to add some for `whatever SERIAL NOT NULL` now that I think about it.

Reply:
Okay, coming back to this `notnull` thing. I'm not actually it sure it should get checked at all. Any default value for a serial field is invalid -- they already have defaults built in. Maybe `getDefaultValueDeclarationSQL` should just return its empty string for any `SERIAL` field?